### ISSUE TITLE
Pay It Forward

### ISSUE THREAD

**Issue by 0xGabi** [1Hive/ideas#47]
# Pay It Forward

I been thinking in ways to bring Aragon and Sourcecreed technologies closer. First I imagine a way to use a token as an incentive mechanism to trade favors between people, like in the movie [“Pay It Forward”](https://es.wikipedia.org/wiki/Pay_It_Forward_(pel%C3%ADcula)). Expanding into this idea and trying to figure out how the incentives could be codify. I arrived at the following: **what if we are able to create a Sourcecreed score from the history of transactions of a token.**

In that sense token holders start having a kind of reputation and skin in the game on the way they use their tokens. This gives tokens a second value proposition other than their monetary value. 

Then if we start calculating a score on top of Aragon we end up with a new way to measure how each token holder interact inside an organization and how their contributions weight in.

- Ex: If you mint a token for a new member of the organization and that new member end-up been a great addition, you also receive the benefits of that decision.

Going forward we can start adapting Sourcecreed algorithm and start leveraging new data inputs from different Aragon apps and how each affect the final score of the members. 

- Ex: We can consider votes on a Voting app as a new input into the score calculation for each token holder. We can go even further and eventually have labels per vote topic and consider the score for each topic and start having something like a baseball card scoring on each topic (thanks @rperez89 for the insight) 

Looking forward for feedback about this idea and let me know if it makes sense for you 🙌

**Comment by sembrestels** [comment]
Not sure if I understood well. Are you proposing to use Aragon DAOs data (transactions from tokens, votings...) to feed sourcecred instead of using a git repo? That sounds as a really cool intersectional project between credao and apiary!

**Comment by 0xGabi** [comment]
Yes, that's probably a simple way to put it. It's interesting to source that data cause most relevant actions on a DAO are linked to a token (usually a Minime).

I thought it as a way to generate a score (reputation) for every entity on the DAO. Once you have that score you can experiment with ways to use that information.

**Comment by yeqbfgxjiq** [comment]
Are you saying that you could have a DAO and it could have two tokens: membership and reputation. When members engage with the DAO, such as via voting, they would create a graph of activity. You could then run SourceCred on the voting app to create a reputation score for members of the DAO. This reputation might be measured naively by saying that creating a vote is 2 points and voting is 1 point. To then enhance this via pagerank you could say that if the vote passes that flows Cred to the person who created the vote, but you'd probably want to have some sort of slashing if the vote fails. Since the weight of Cred that one contributes towards these votes is determined by reputation, overtime those who participate in voting as well as those who also create votes that people approve in would earn more reputation than those who don't. Is this kind of what you're thinking?
